File: Python/22. Generate Parentheses.py
from typing import List

class Solution:
    def generateParenthesis(self, n: int) -> List[str]:
        if n==0:
            return [""]
        ans = []
        for i in range(n):
            for left in self.generateParenthesis(i):
                for right in self.generateParenthesis(n-1-i):
                    ans.append("("+left+")"+right)
        return ans

if __name__ == "__main__":
    n = 3
    result =  Solution().generateParenthesis(n)
    print(result)

File: CPP/22. Generate Parentheses.cpp
#include <iostream>
#include <vector>
#include <string>
#include "printFunctions.hpp"

using namespace std;

class Solution {
public:
	vector<string> generateParenthesis(int n) {
		if (n == 0) {
			return { "" };
		}
		vector<string> ans = {};
		for (int i = 0;i < n;i++) {
			for (string left : generateParenthesis(i)) {
				for (string right : generateParenthesis(n - 1 - i)) {
					ans.push_back("(" + left + ")" + right);
				}
			}
		}
		return ans;
	}
};

int main() {
	Solution sol;
	int n = 3;
	vector<string> result = sol.generateParenthesis(n);
	printVector(result);
}